handlers/webchat/webchat.go

```diff
@@ -104,6 +122,78 @@ func (h *handler) receiveMessage(ctx context.Context, channel Channel, w http.Re
 	return handlers.WriteMsgsAndResponse(ctx, h, []Msg{msg}, w, r)
 }
 
+func (h *handler) chatHistory(ctx context.Context, channel Channel, w http.ResponseWriter, r *http.Request) ([]Event, error) {
+	payload := &userPayload{}
+	err := handlers.DecodeAndValidateJSON(payload, r)
+	if err != nil || payload.UserToken == "" {
+		return nil, handlers.WriteAndLogRequestIgnored(ctx, h, channel, w, r, "Ignoring request, no contact token")
+	}
+
+	urnString, err := urnFromToken(payload.UserToken)
+	if err != nil {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, channel, w, r, err)
+	}
+
+	urn, errURN := urns.Parse(urnString)
+	if errURN != nil {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, channel, w, r, errURN)
+	}
+
+	contact, errGetContact := h.Backend().GetContact(ctx, channel, urn, "", "")
+	if errGetContact != nil {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, channel, w, r, errGetContact)
+	}
+
+	msgs, err := h.Backend().GetContactMessages(channel, contact)
+	if err != nil {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, channel, w, r, err)
+	}
+
+	// the list of data we will return in our response
+	responseMsgs := make([]*webChatMessagePayload, 0)
+	for _, msg := range msgs {
+		origin := "user"
+		msgDirection := msg.(*rapidpro.DBMsg).Direction_
+		if msgDirection == "O" {
+			origin = "ws"
+		}
+		responseMsg := &webChatMessagePayload{
+			Message:     msg.Text(),
+			Origin:      origin,
+			Metadata:    nil,
+			Attachments: nil,
+		}
+
+		metadata := make(map[string]interface{}, 0)
+
+		if len(msg.QuickReplies()) > 0 {
```

Comment:
Quick question: the endpoint `history` will be called just when the user refreshes the page or when it's initializing the webchat with a user already on local storage, right? If so, I don't think we will need to show again the `quick_replies`, `receive_attachment` and `sharing_config`.

Reply:
I believe we need it in the case when the last message was with quick replies, or so on, but the user hasn't answered before the page refresh. We need to load it again to show it to the user.